instance/models/openedx_instance.py

```diff
@@ -146,6 +147,21 @@ def studio_domain(self):
             return self.internal_studio_domain
 
     @property
+    def studio_domain_nginx_regex(self):
+        """
+        Regex that matches either the internal or the external Studio URL.
+
+        This is exclusively meant for the Ansible variable CMS_HOSTNAME to configure the nginx
+        server_name regex to match the Studio domains.
+        """
+        if self.external_studio_domain:
+            domains = [self.external_studio_domain, self.internal_studio_domain]
+        else:
+            domains = [self.internal_studio_domain]
+        choices = '|'.join(map(re.escape, domains))  # pylint: disable=bad-builtin
```

Comment:
@itsjeyd Alright, thanks for the explanation - yes, let's try to limit the exclusions to cases where we can't do it any other way. Besides making them more manageable, it will also help with having a consistent style. For this particular one, I've added it to the list of OC-1572.